### ISSUE TITLE
The number of tokens to keep from the initial prompt is greater than the context length.

### ISSUE THREAD

**Issue by itwone** [logancyang/obsidian-copilot#962]
![bug](https://github.com/user-attachments/assets/c3bf6677-8088-4d29-9113-f1c130c58a85)
![bug2](https://github.com/user-attachments/assets/58957036-d24c-4127-b1d2-4534f0e9ac3a)
- [ ] Disable all other plugins besides Copilot **(required)**
- [ ] Screenshot of note + Copilot chat pane + dev console added **(required)**

Copilot version:

(Bug report without the above will be closed)

**Describe how to reproduce**
A clear and concise description of what the bug is. Clear steps to reproduce the behavior

**Expected behavior**

**Screenshots**
Add screenshots to help explain your problem. Please turn on debug mode in Copilot settings, turn off other plugins to leave only Copilot dev messages as necessary.

**Additional context**

**Comment by logancyang** [comment]
Essentially we need the context length info from the model somehow, either provided by the user or the API.

**Comment by logancyang** [comment]
Thanks for reporting this, and sorry for the slow reply.

Copilot v3 is in code freeze, so we're closing this as not planned. v4 is a ground-up rewrite and is launching soon, with a very different architecture around agent mode.

If this is still an issue for you in v4 once it lands, please open a new issue and we'll take a fresh look. 🙏
